File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    static void dfs(List<List<Integer>> graph, boolean[] visited, int node) {
        // System.out.println(node);
        visited[node] = true;
        
        for (int nxt: graph.get(node)) {
            if (!visited[nxt]) {
                dfs(graph, visited, nxt);
                visited[nxt] = true;
            }
        }
    }
    
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        List<List<Integer>> graph = new ArrayList<>();
        
        for (int i=0; i<n; i++) {
            graph.add(new ArrayList<>());
        }
        
        for (int i=0; i<n; i++) {
            for (int j=0; j<n; j++) {
                if (computers[i][j]==1) {
                    graph.get(i).add(j);
                    graph.get(j).add(i);
                }
            }
        }
        
        
        boolean[] visited = new boolean[n];
        for (int i=0; i<n; i++) {
            if (!visited[i]) {
                answer++;
                dfs(graph, visited, i);
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.kt
class Solution {
    fun dfs(graph: ArrayList<ArrayList<Int>>, visited: BooleanArray, node: Int) {
        visited[node]=true
        
        for (nxt in graph[node]) {
            if (!visited[nxt]) {
                dfs(graph, visited, nxt)
            }
        }
    }
    
    fun solution(n: Int, computers: Array<IntArray>): Int {
        var answer = 0
        var graph = ArrayList<ArrayList<Int>>()
        
        for(i in 0 until n) {
            graph.add(ArrayList())
        }
        
        for (i in 0 until n) {
            for (j in 0 until n) {
                if (computers[i][j]==1) {
                    graph[i].add(j)
                }
            }
        }
        
        var visited = BooleanArray(n)
        
        for (i in 0 until n) {
            if (!visited[i]) {
                dfs(graph, visited, i)
                answer++
            }
        }
        
        return answer
    }
}